Intent adaptations for multiple homes

TAKeanice opened the pull request:
I tried making the change as unintrusive as possible, and all existing automations should stay as they are.

Even if no home name provided, as long as the item name is unique, all shortcuts will work.

There is an edge case where two homes have the same item name, and no home name is provided. It should work in theory, the code is supposed to ask for disambiguation then, but I have not tested it yet.

TAKeanice committed:
add intent parameter for home

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
remove nonexisting "new group"

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
Some steps towards multi-home compatible intents

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
larger refactoring of intents (WIP)

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
adapt getItemStateIntentHandler

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
adapt all intents for multiple homes

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
warning when renaming homes

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
add warning about changing home names

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice commented:
I thought about that, too, as homes are already identified by a UUID internally. As you can see, I always use it when executing the intent. Not sure whether we can have nice home names while internally identifying the homes with UUIDs.

digitaldan commented:
Gotcha, sorry i missed that.  So i'm not at all familiar with intents, but after doing a little reading, it sounds like you can pass custom objects instead of strings.  Can we change "home" to a INObject?  So in the XCode UI:

<img width="1105" height="437" alt="image" src="https://github.com/user-attachments/assets/905ef8f4-86b9-4339-b9e5-78814c555559" />

Where the type `Home` is defined as:

<img width="1092" height="414" alt="image" src="https://github.com/user-attachments/assets/2cdb45a3-58fa-48ba-bdef-e1b4b064d540" />

I think those are the default/required INObject fields (display and identifier) , and  would work nicely for us without having to add any other custom fields?

TAKeanice commented:
I will try and adapt my implementation if possible, I agree it would be more elegant.

TAKeanice committed:
use home intent object to make identification easier

also, remove static part from item cache to profit from actor isolation (networkTrackers array always produced data races)

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice commented:
> you can pass custom objects instead of strings. Can we change "home" to a INObject? So in the XCode UI:

Done that and it works now (tests are still not fixed though)
